platform/android/java_godot_lib_jni.cpp

```diff
@@ -158,6 +158,7 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_setup(JNIEnv *env, jc
 	}
 
 	java_class_wrapper = memnew(JavaClassWrapper(godot_java->get_activity()));
+	ClassDB::register_class<JNISingleton>();
```

Comment:
This will mess up API portability. You have to register it in `platform/android/api/api.cpp` and provide a stub version for other platforms. See a6105c8ea0bdae9d56d1943409d89cc71288306a for reference on how it was done for `JavaClassWrapper`.